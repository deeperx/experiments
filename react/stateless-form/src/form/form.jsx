import React from 'react'

/*
 * This form automatically collects its inputs
 * Now, it only supports input fields
 */
const Form = props => {
  let details = {}

  const _onChange = e => {
    const { name, value } = e.target
    details[name] = value
  }

  const _onSubmit = e => {
    e.preventDefault()
    props.onSubmit(details)
  }

  return (
    <form onSubmit={ _onSubmit } onChange={ _onChange }>
      { props.children }
    </form>
  )
}

export default Form
