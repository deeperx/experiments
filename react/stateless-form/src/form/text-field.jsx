import React from 'react'

const TextField = ({ title, field, defaultValue = '' }) => (
  <p>
    <label style={{
      textTransform: "capitalize",
      marginRight: 10
    }}>
      { title }:
    </label>

    <input
      type="text"
      name={ field }
      defaultValue={ defaultValue }
    />
  </p>
)

export default TextField
