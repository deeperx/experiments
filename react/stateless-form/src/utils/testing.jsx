import renderer from 'react-test-renderer'

export const expectToMatchSnapshot = el =>
  expect(
    renderer.create(el).toJSON()
  ).toMatchSnapshot()
