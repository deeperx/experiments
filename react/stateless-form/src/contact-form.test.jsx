import React from 'react'
import ReactDOM from 'react-dom'
import { expectToMatchSnapshot } from './utils/testing'

import { Form, TextField } from './form/'
import ContactForm from './contact-form.jsx'

it('renders correctly', expectToMatchSnapshot(
  <ContactForm />
))

// functional test: should be separated
it('gets the correct details on submit', () => {
  const onSubmit = jest.fn()
  const form = mount(<ContactForm onSubmit={ onSubmit } />)

  inputSimulateChangeMulti([
    { wrapper: form, field: 'name', value: 'inanc' },
    { wrapper: form, field: 'lastname', value: 'gumus' },
    { wrapper: form, field: 'email', value: 'inanc@pixenka.com' }
  ])

  form.simulate('submit')

  expect(onSubmit).toBeCalledWith(
    { name: 'inanc', lastname: 'gumus', email: 'inanc@pixenka.com' }
  )
})

// helpers
const inputSimulateChangeMulti = els => els.forEach(
  el => inputSimulateChange(el))

const inputSimulateChange = ({ wrapper, field, value }) => {
  wrapper.find(`input[name="${ field }"]`)
         .first()
         .simulate('change', {
           target: { name: field, value }
         })
}
