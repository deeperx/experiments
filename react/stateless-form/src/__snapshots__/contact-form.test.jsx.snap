exports[`test renders correctly 1`] = `
<form>
  <p>
    <span
      style={
        Object {
          "marginRight": 10,
          "textTransform": "capitalize",
        }
      }>
      Your name
      :
    </span>
    <input
      name="name"
      type="text" />
  </p>
  <p>
    <span
      style={
        Object {
          "marginRight": 10,
          "textTransform": "capitalize",
        }
      }>
      Your lastname
      :
    </span>
    <input
      name="lastname"
      type="text" />
  </p>
  <p>
    <span
      style={
        Object {
          "marginRight": 10,
          "textTransform": "capitalize",
        }
      }>
      Your email
      :
    </span>
    <input
      name="email"
      type="text" />
  </p>
</form>
`;

exports[`undefined 1`] = `
<form
  onChange={[Function]}
  onSubmit={[Function]}>
  <p>
    <label
      style={
        Object {
          "marginRight": 10,
          "textTransform": "capitalize",
        }
      }>
      Your name
      :
    </label>
    <input
      defaultValue=""
      name="name"
      type="text" />
  </p>
  <p>
    <label
      style={
        Object {
          "marginRight": 10,
          "textTransform": "capitalize",
        }
      }>
      Your lastname
      :
    </label>
    <input
      defaultValue=""
      name="lastname"
      type="text" />
  </p>
  <p>
    <label
      style={
        Object {
          "marginRight": 10,
          "textTransform": "capitalize",
        }
      }>
      Your email
      :
    </label>
    <input
      defaultValue=""
      name="email"
      type="text" />
  </p>
  <input
    type="submit"
    value="Submit" />
  <input
    type="reset"
    value="Clear" />
</form>
`;
